File: rbz-tasks/GardenFence/GardenFence.cs
using System;
using TaskSystem;

namespace GardenFence
{ 
    public class GardenFenceTaskProcessor: ITaskProcessor
    {
        public void Process()
        {
            Console.WriteLine($"Der geheime Code lautet: {Geheim("HABEAMSONNTAGKEINEZEITMUSSLERNENSORRY")}");
            Console.WriteLine($"Der nicht mehr geheime Code lautet: {NichtGeheim("LEEMTEOAEENLMTITROLNIBRIVRNKGLASIDEEBHE")}");
            Console.WriteLine($"Der nicht mehr geheime Code lautet: {NichtGeheim("AMTWCFLEDERTNWITNEASMITOHALNIESEZESUDNU")}");
        
            Console.WriteLine("Gebe dein Text zur Verschlüsselung ein...");
            var encode = Console.ReadLine() ?? throw new InvalidOperationException();
            Console.WriteLine($"Deine Verschlüsselung: {Geheim(encode)}");
        
            Console.WriteLine("Gebe dein Text zur Entschlüsselung ein...");
            var decode = Console.ReadLine() ?? throw new InvalidOperationException();
            Console.WriteLine($"Deine Entschlüsselung: {NichtGeheim(decode)}");
        }

        private static string Geheim(string input)
        {
            var first = "";
            var second = "";
            for (var i = 0; i < input.Length; i++)
            {
                var character = input[i];
                if (i % 2 == 0)
                {
                    first = $"{first}{character.ToString()}";
                }
                else
                {
                    second = $"{second}{character.ToString()}";
                }
            }
            return $"{first}{second}";
        }

        private static string NichtGeheim(string input)
        {
            var inputLength = input.Length / 2;
            var length = inputLength + input.Length % 2;

            var output = "";
            for (var i = 0; i < length; i++)
            {
                output = $"{output}{input[i].ToString()}";

                if (i < inputLength)
                {
                    output = $"{output}{input[length + i].ToString()}";
                }
            }
            return output;
        }
        
        public string GetName()
        {
            return "GardenFence";
        }
    }
}

File: Kotlin/rbz-task/src/main/kotlin/de/rbztechnik/task/tasks/GardenFence.kt
package de.rbztechnik.task.tasks

import de.rbztechnik.task.TaskProcessor

class GardenFence: TaskProcessor {

    override fun process() {
        println("Der geheime Code lautet: ${decode("HABEAMSONNTAGKEINEZEITMUSSLERNENSORRY")}")
        println("Der nicht mehr geheime Code lautet: ${encode("LEEMTEOAEENLMTITROLNIBRIVRNKGLASIDEEBHE")}")
        println("Der nicht mehr geheime Code lautet: ${encode("AMTWCFLEDERTNWITNEASMITOHALNIESEZESUDNU")}")
    }

    private fun decode(message: String): String {
        var first = "";
        var second = "";
        for (i in message.indices) {
            val character = message[i]
            if (i % 2 == 0) {
                first = first.plus(character)
            } else {
                second = second.plus(character)
            }
        }
        return first.plus(second)
    }

    private fun encode(message: String): String {
        val inputLength = message.length / 2
        val length = inputLength + message.length % 2

        var output = ""
        for (i in 0 until length) {
            output = output.plus(message[i])

            if (i < inputLength) {
                output = output.plus(message[length + i])
            }
        }
        return output;
    }

    override fun getName(): String = "GardenFence"
}